File: refactortalentscrm_ui/src/services/AssignmentService.js
import http from "../http-common";

class AssignmentService {
  getAllAssignments(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/assignment/assignments`, searchDTO);
  }

  get(assignmentId) {
    return this.getRequest(`/assignment/${assignmentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/assignment?field=${matchData}`, null);
  }

  addAssignment(data) {
    return http.post("/assignment/addAssignment", data);
  }

  update(data) {
  	return http.post("/assignment/updateAssignment", data);
  }
  
  uploadImage(data,assignmentId) {
  	return http.postForm("/assignment/uploadImage/"+assignmentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AssignmentService();

File: src/main/java/com/refactortalentscrm/service/AssignmentService.java
package com.refactortalentscrm.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.refactortalentscrm.domain.Assignment;
import com.refactortalentscrm.dto.AssignmentDTO;
import com.refactortalentscrm.dto.AssignmentSearchDTO;
import com.refactortalentscrm.dto.AssignmentPageDTO;
import com.refactortalentscrm.dto.AssignmentConvertCriteriaDTO;
import com.refactortalentscrm.service.GenericService;
import com.refactortalentscrm.dto.common.RequestDTO;
import com.refactortalentscrm.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AssignmentService extends GenericService<Assignment, Integer> {

	List<Assignment> findAll();

	ResultDTO addAssignment(AssignmentDTO assignmentDTO, RequestDTO requestDTO);

	ResultDTO updateAssignment(AssignmentDTO assignmentDTO, RequestDTO requestDTO);

    Page<Assignment> getAllAssignments(Pageable pageable);

    Page<Assignment> getAllAssignments(Specification<Assignment> spec, Pageable pageable);

	ResponseEntity<AssignmentPageDTO> getAssignments(AssignmentSearchDTO assignmentSearchDTO);
	
	List<AssignmentDTO> convertAssignmentsToAssignmentDTOs(List<Assignment> assignments, AssignmentConvertCriteriaDTO convertCriteria);

	AssignmentDTO getAssignmentDTOById(Integer assignmentId);







}





